[css-masonry] Update Intrinsic Sizing Rows 003 FR Expected / Ref Test Case

stwrt opened the pull request:
#### 7abc3005b876577618d227b274bfee64e7386a87
<pre>
[css-masonry] Update Intrinsic Sizing Rows 003 FR Expected / Ref Test Case
<a href="https://bugs.webkit.org/show_bug.cgi?id=282992">https://bugs.webkit.org/show_bug.cgi?id=282992</a>
<a href="https://rdar.apple.com/problem/139727829">rdar://problem/139727829</a>

Reviewed by Sammy Gill.

Updated the font to Ahem to tackle the computed width sizing issue of &quot;3 3&quot;. Added colors to the different items to catch ordering issues.

* Grid 2

Item 5 should be a height of 2ch.

* Grid 4

Remove item 3 span, because it was causing width issues with the grid.

* Grid 5

Remove item 3 span, because it was causing width issues with the grid.

* Grid 6

Remove item&apos;s 2 &amp; 3 spans; they were causing width issues.
Remove extra 5 that is not in the original test case.
Removed hidden items.

* Grid 7

Remove item&apos;s 2 &amp; 3 spans.
Removed hidden items.
Simplified item 5 to just span 4.

* Grid 8

Remove item&apos;s 2 &amp; 3 spans.
Removed hidden items.
Simplified item 5 to just span 4 and have a height of 6ch.

* Grid 9

Remove item&apos;s 2 &amp; 3 spans.
Removed hidden items.
Simplified item 5 to just span 3 and have a height of 6ch.

* LayoutTests/TestExpectations:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/intrinsic-sizing/masonry-intrinsic-sizing-rows-003-fr-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/intrinsic-sizing/masonry-intrinsic-sizing-rows-003-fr-ref.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/intrinsic-sizing/masonry-intrinsic-sizing-rows-003-fr.html:

Canonical link: <a href="https://commits.webkit.org/286749@main">https://commits.webkit.org/286749@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e3b2b05af7c28c150fc54b84eef5daba6e32ee59

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/76994 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/56029 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/29909 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/81542 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/28273 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/65177 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/4325 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/60340 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/18413 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/80061 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/50282 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/66095 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/40652 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/47684 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/23592 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/26599 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/68804 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/23921 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/82977 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/4374 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/2933 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/68611 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/4529 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/66068 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/67862 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/16915 "Built successfully and passed tests") | [  ~~🧪 vision-wk2~~](https://ews-build.webkit.org/#/builders/126/builds/11851 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/119/builds/9935 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/4320 "Built successfully") | | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/125/builds/4340 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/129/builds/7775 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/6099 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

sammygill commented on LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/intrinsic-sizing/masonry-intrinsic-sizing-rows-003-fr-expected.html:
```diff
@@ -48,11 +74,11 @@
-  <item style="grid-column: span 2">2 2</item>
-  <item style="grid-column: span 2">3 3</item>
-  <item>4</item>
-  <item style="grid-column: span 2">5 5</item>
```
Probably just needs to be `style="height: 2ch"` and everything else can be the same

stwrt replied:
Tested this out; we can just update number 5.

sammygill commented on LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/intrinsic-sizing/masonry-intrinsic-sizing-rows-003-fr-expected.html:
```diff
@@ -70,72 +96,52 @@
-  <item class="hidden" style="grid-area: 3/3">0</item>
-  <item class="hidden" style="grid-area: 2/1">0 0</item>
-  <item class="hidden" style="grid-area: 3/1">0 0</item>
 </grid>
```
We should be careful about removing the hidden items since they can affect track sizing. I think everything works out here but it's possible there are subtle differences in the test cases. The only thing you had to change here was to make sure "3 3" was a span of 1 to get rid of the extra spacing you were seeing

sammygill commented on LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/intrinsic-sizing/masonry-intrinsic-sizing-rows-003-fr-expected.html:
```diff
@@ -70,72 +96,52 @@
-  <item style="height:4ch; grid-area: 2/1/4/auto">5 5</item>
-
-  <item class="hidden" style="grid-area: 2/1">0 0</item>
-  <item class="hidden" style="grid-area: 3/2">0 0</item>
```
I don't think it's the hidden items that are causing the gap here. I think it is the "span 2" for the 3 3 item

stwrt replied:
What are you proposing to do here?

stwrt replied:
Added the hidden items back.

sammygill commented on LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/intrinsic-sizing/masonry-intrinsic-sizing-rows-003-fr-expected.html:
```diff
@@ -70,72 +96,52 @@
   <item>4</item>
-  <item style="height:6ch; grid-area: 1/3/5/4">5 5</item>
-
-  <item class="hidden" style="grid-area: 1/3">0 0</item>
-  <item class="hidden" style="grid-area: 4/3">0 0</item>
```
Don't think these need to be removed just moved to column 2?

stwrt replied:
Item 5 is taking up all of column 2.

sammygill replied:
I think it's fine since these items are hidden and are just there to influence track sizing? You may need to double check that the actual item(s) that are placed in the same track are placed explicitly in case they end up moving due to the hidden items being forced there

sammygill commented on LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/intrinsic-sizing/masonry-intrinsic-sizing-rows-003-fr-expected.html:
```diff
@@ -70,72 +96,52 @@
   <item>1</item>
-  <item style="grid-column: span 2">2 2</item>
-  <item style="grid-column: span 2">3 3</item>
+  <item>2 2</item>
+  <item>3 3</item>
   <item>4</item>
-  <item style="height:6ch; grid-area: 1/3/4/4">5 5</item>
-
-  <item class="hidden" style="grid-area: 1/3">0 0</item>
-  <item class="hidden" style="grid-area: 4/3">0 0</item>
-  <item class="hidden" style="height:6ch; grid-area: 2/3/5/4">0 0</item>
```
I was having a hard time figuring out what was going on with this one, but what I *think* we need to do it:
- remove the span 2 from the second and third items like yo are doing
- remove the third column that is being created from the placement properties

AFAICT the tracks should be large enough to hold the content and not cause wrapping so the 5 5 item should be placed right next to 1 and adding a third column introduces that spacing. I don't *think* there is a reason to have a 3rd column in this test case

sammygill reviewed (commented):
Are these test expectations actually correct in reference to the spec change about `masonry-auto-flow`? My understanding is that property no longer exists so (unless you use the order property) items are placed with respect to the DOM order.

While I was going through some of these changes I found that the new expectations don't quite match with my current understanding. 


For example, when looking at grid 6, "1," comes first in DOM order so I would expect that to get placed in the first row. However, the current updated expectation still places it in row 4. Is my understanding of the new spec wrong or does this still require further test expectation updating?

sammygill commented on LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/intrinsic-sizing/masonry-intrinsic-sizing-rows-003-fr-expected.html:
```diff
@@ -71,18 +97,17 @@
   <item class="hidden" style="grid-area: 2/1">0 0</item>
-  <item class="hidden" style="grid-area: 3/2">0 0</item>
-</grid>
+  <item class="hidden" style="grid-area: 3/2">0 0</item></grid>
```
`</grid>` should be on its own line

sammygill commented on LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/intrinsic-sizing/masonry-intrinsic-sizing-rows-003-fr-ref.html:
```diff
@@ -71,18 +97,17 @@
   <item class="hidden" style="grid-area: 2/1">0 0</item>
-  <item class="hidden" style="grid-area: 3/2">0 0</item>
-</grid>
+  <item class="hidden" style="grid-area: 3/2">0 0</item></grid>
```
`</grid>` should be on its own line

sammygill reviewed (commented):
I think we need to update the commit message here to reflect the new changes

stwrt committed:
[css-masonry] Update Intrinsic Sizing Rows 003 FR Expected / Ref Test Case
https://bugs.webkit.org/show_bug.cgi?id=282992
rdar://problem/139727829

Reviewed by Sammy Gill.

Updated the font to Ahem to tackle the computed width sizing issue of "3 3". Added colors to the different items to catch ordering issues.

* Grid 2

Item 5 should be a height of 2ch.

* Grid 4

Remove item 3 span, because it was causing width issues with the grid.

* Grid 5

Remove item 3 span, because it was causing width issues with the grid.

* Grid 6

Remove item's 2 & 3 spans; they were causing width issues.
Remove extra 5 that is not in the original test case.
Removed hidden items.

* Grid 7

Remove item's 2 & 3 spans.
Removed hidden items.
Simplified item 5 to just span 4.

* Grid 8

Remove item's 2 & 3 spans.
Removed hidden items.
Simplified item 5 to just span 4 and have a height of 6ch.

* Grid 9

Remove item's 2 & 3 spans.
Removed hidden items.
Simplified item 5 to just span 3 and have a height of 6ch.

* LayoutTests/TestExpectations:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/intrinsic-sizing/masonry-intrinsic-sizing-rows-003-fr-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/intrinsic-sizing/masonry-intrinsic-sizing-rows-003-fr-ref.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/intrinsic-sizing/masonry-intrinsic-sizing-rows-003-fr.html:

Canonical link: https://commits.webkit.org/286749@main
